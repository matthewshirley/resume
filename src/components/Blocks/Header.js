import React from 'react';
import PropTypes from 'prop-types';
import { Box, Flex, Text } from '..';

// eslint-disable-next-line react/prop-types
const renderCompany = ({ website, company }) => {
  if (website) {
    return (
      <a href={website} target="_blank" rel="noopener noreferrer">
        {company}
      </a>
    );
  }

  return (company);
};

// eslint-disable-next-line
const renderPositionData = ({ company, position, website, location }) => (
  <Box>
    <Text fontSize={1} m={0} mr={1} bold>
      {position}
    </Text>
    <Text as="small" fontSize={0} m={0}>
      {renderCompany({ website, company })}
      {location && ` - ${location}`}
    </Text>
  </Box>
);

// eslint-disable-next-line react/prop-types
const renderDates = ({ startDate, endDate }) => {
  const isPresent = endDate === false || endDate === 'Jan 1900'; // TODO: Multiple GraphQL types?
  const formattedEndDate = isPresent ? 'Present' : endDate;

  return (
    <Text fontSize={0} m={0}>
      {`${startDate} - ${formattedEndDate}`}
    </Text>
  );
};

const Header = ({
  company, position, website, location, startDate, endDate,
}) => (
  <Flex justifyContent="space-between" mb={2}>
    {renderPositionData({
      company, position, website, location,
    })}
    {renderDates({ startDate, endDate })}
  </Flex>
);

Header.propTypes = {
  company: PropTypes.string.isRequired,
  position: PropTypes.string.isRequired,
  startDate: PropTypes.string.isRequired,
  endDate: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.bool,
  ]).isRequired,
  location: PropTypes.string,
  website: PropTypes.string,
};

Header.defaultProps = {
  location: '',
  website: '',
};

export default Header;
