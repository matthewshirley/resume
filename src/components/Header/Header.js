import React from 'react';
import PropTypes from 'prop-types';

import Flexbox from '../common/Flexbox';
import { Heading } from '../common/Text';
import { EmailLink, PhoneLink } from '../common/Link';

const Header = ({ name, email, phone }) => (
  <Flexbox as="header" gridArea="header" justifyContent="space-between" mb={{ sm: 5, lg: 6, print: 6 }} flexDirection={{ sm: 'column', lg: 'row', print: 'row' }}>
    <Heading>{name}</Heading>

    {(email || phone) && (
    <Flexbox flexDirection="column" textAlign={{ sm: 'left', lg: 'right', print: 'right' }}>
      <EmailLink emailAddress={email} />
      <PhoneLink phoneNumber={phone} />
    </Flexbox>
    )}
  </Flexbox>
);

Header.propTypes = {
  name: PropTypes.string.isRequired,
  email: PropTypes.string,
  phone: PropTypes.string,
};

Header.defaultProps = {
  email: null,
  phone: null,
};

export default Header;
