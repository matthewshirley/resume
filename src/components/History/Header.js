import React from 'react';
import PropTypes from 'prop-types';

import Box from '../common/Box';
import Text from '../common/Text';
import Link from '../common/Link';

const Header = ({
  company, position, location, website,
}) => (
  <Box>
    <Text m={0} mr={1} bold>
      {position}
    </Text>
    <Text as="small" m={0}>
      <Link href={website}>{company}</Link>
      {location && ` - ${location}`}
    </Text>
  </Box>
);

Header.propTypes = {
  company: PropTypes.string.isRequired,
  position: PropTypes.string.isRequired,
  location: PropTypes.string,
  website: PropTypes.string,
};

Header.defaultProps = {
  location: null,
  website: null,
};

export default Header;
