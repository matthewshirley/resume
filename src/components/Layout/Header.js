import React from 'react';
import PropTypes from 'prop-types';

import { Box, Flex, Text } from '@mattshirley/design';

import headerGrid from './header.module.css';

/**
 * The page header outlining basic information (name, phone number).
 *
 * @param {String} name
 * @param {String} label
 * @param {String} email
 * @param {String} phone
 */
const PageHeader = ({
  name, label, email, phone,
}) => (
  <Box as="header" className={headerGrid.container} mb={5}>
    <Box className={headerGrid.Name}>
      <Text as="h1" color="primary" m={0}>{name}</Text>
      <Text as="small" m={0}>{label}</Text>
    </Box>
    <Box className={headerGrid.Contact}>
      <Flex flexDirection="column">
        <a href={`mailto:${email}`}>
          <Text as="small" m={0}>{email}</Text>
        </a>
        <a href={`tel:${phone}`}>
          <Text as="small" m={0}>{phone}</Text>
        </a>
      </Flex>
    </Box>
  </Box>
);

PageHeader.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string,
  email: PropTypes.string,
  phone: PropTypes.string,
};

PageHeader.defaultProps = {
  label: '',
  email: '',
  phone: '',
};

export default PageHeader;
