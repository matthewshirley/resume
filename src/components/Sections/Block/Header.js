import React from 'react';
import PropTypes from 'prop-types';

import { Text } from '@mattshirley/design/src/components/Text';
import { Box } from '@mattshirley/design/src/components/Layout/Box';
import { Flex } from '@mattshirley/design/src/components/Layout/Flex';

const Header = ({
  company, position, website, location, startDate, endDate,
}) => {
  const formattedStartDate = startDate;
  const formattedEndDate = endDate || 'Present';

  return (
    <Flex justifyContent="space-between" mb={2}>
      <Box>
        <Text fontSize={1} bold>
          {position}
        </Text>
        <Text as="small" fontSize={0}>
          <a href={website} target="_blank" rel="noopener noreferrer">
            {company}
          </a>
          {location && ` - ${location}`}
        </Text>
      </Box>
      <Text fontSize={0}>
        {formattedStartDate}
        {' '}
        -
        {' '}
        {formattedEndDate}
      </Text>
    </Flex>
  );
};

Header.propTypes = {
  company: PropTypes.string.isRequired,
  position: PropTypes.string.isRequired,
  website: PropTypes.string.isRequired,
  startDate: PropTypes.string.isRequired,
  endDate: PropTypes.oneOf([
    PropTypes.bool, PropTypes.string]).isRequired,
  location: PropTypes.string,
};

Header.defaultProps = {
  location: '',
};

export default Header;
