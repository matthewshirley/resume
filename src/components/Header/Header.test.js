import React from 'react';
import renderer from 'react-test-renderer';

import Header from './Header';

describe('Header', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Header name="Test Name" phone="111 111 1111" email="test@user.com" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without a phone or email', () => {
    const tree = renderer
      .create(<Header name="Test Name" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
