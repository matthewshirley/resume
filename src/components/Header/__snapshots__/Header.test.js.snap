// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="sc-AxjAm sc-AxirZ liihYQ"
>
  <h1
    className="sc-AxiKw sc-AxhCb dGbZkM"
    color="primary"
    fontSize={5}
  >
    Test Name
  </h1>
  <div
    className="sc-AxjAm sc-AxirZ kYfuYW"
  >
    <a
      className="sc-AxiKw bFEIqf"
      fontSize={1}
      href="mailto:test@user.com"
    >
      test@user.com
    </a>
    <a
      className="sc-AxiKw bFEIqf"
      fontSize={1}
      href="tel:111 111 1111"
    >
      111 111 1111
    </a>
  </div>
</header>
`;

exports[`Header renders correctly without a phone or email 1`] = `
<header
  className="sc-AxjAm sc-AxirZ liihYQ"
>
  <h1
    className="sc-AxiKw sc-AxhCb dGbZkM"
    color="primary"
    fontSize={5}
  >
    Test Name
  </h1>
</header>
`;
